### PR TITLE
NPC barber talk function to make you look poppin fresh

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Features "Allows NPC to change your hairstyle or beard style"```

#### Purpose of change

@I-am-Erk  asked me to do this.

#### Describe the solution
Talk function "barber_hair" or "barber_shave" will open up a uilist slection menu, to choose your new style.
It will remove the current one, add the new one, all instantly.
Meant to be used in conjunction with the existing buy_haircut or buy_shave talk functions which implement an activity and morale boost after its done.

#### Describe alternatives you've considered
N/A

#### Additional context
I have not added any NPC as an example of this, as thats gonna be what Erk does, but its trivial to add the talk fnction effect to any existing dialogue to test.

### CODE REVIEW

**Comment by I-am-Erk** [comment]
As soon as this is merged, I will make it possible to get your hair cut properly in the refugee center and tacoma commune.

**Comment by EirenexTheDragon** [comment]
"> Talk function "barber_hair" or "barber_shave" will open up a uilist slection menu, to choose your new style."

Does that mean we can get our beardless character shaved into a fullbeard character?

**Comment by I-am-Erk** [comment]
You sure can. I plan to add some jokes about that.

**Comment by Night-Pryanik** [comment]
CJ approves!

изображение

File: src/npctalk_funcs.cpp [hunk 1]
```diff
@@ -499,6 +500,50 @@ void talk_function::give_all_aid( npc &p )
     }
 }
 
+static void generic_barber( std::string mut_type )
+{
+    uilist hair_menu;
+    std::string menu_text = "";
```
Comment (Night-Pryanik) [inline src/npctalk_funcs.cpp]:
```suggestion
    std::string menu_text;
```

File: src/npctalk_funcs.cpp [hunk 2]
```diff
@@ -499,6 +500,50 @@ void talk_function::give_all_aid( npc &p )
     }
 }
 
+static void generic_barber( std::string mut_type )
```
Comment (ZhilkinSerg) [inline src/npctalk_funcs.cpp]:
```suggestion
static void generic_barber( const std::string &mut_type )
```

File: src/npctalk_funcs.cpp [hunk 3]
```diff
@@ -499,6 +500,50 @@ void talk_function::give_all_aid( npc &p )
...
+        hair_menu.addentry( index, true, MENU_AUTOASSIGN, elem.obj().name() );
+    }
+    hair_menu.query();
+    int choice = hair_menu.ret;
+    if( choice != 0 ) {
+        if( g->u.has_trait( cur_hair ) ) {
+            g->u.remove_mutation( cur_hair, true );
+            g->u.set_mutation( hair_muts[ choice - 1 ] );
+            add_msg( m_info, _( "You get a trendy new cut!" ) );
+        }
```
Comment (Night-Pryanik) [inline src/npctalk_funcs.cpp]: Even if player chooses something in the menu, it will do nothing in case he doesn't have any other cut?

Comment (davidpwbrown) [reply src/npctalk_funcs.cpp]: Yeah changed that to always do the cut, but remove cur_hair only if it was set.

**Pull request merged**